components/tidb_query/src/codec/mysql/time/mod.rs

```diff
@@ -925,32 +925,23 @@ pub trait TimeEncoder: NumberEncoder {
     }
 }
 
-impl Time {
-    /// `decode` decodes time encoded by `encode_time` for Chunk format.
-    pub fn decode(data: &mut BytesSlice<'_>) -> Result<Time> {
-        let year = i32::from(number::decode_u16(data)?);
-        let (month, day, hour, minute, second) = if data.len() >= 5 {
-            (
-                u32::from(data[0]),
-                u32::from(data[1]),
-                u32::from(data[2]),
-                u32::from(data[3]),
-                u32::from(data[4]),
-            )
-        } else {
-            return Err(Error::unexpected_eof());
-        };
-        *data = &data[5..];
-        let nanoseconds = 1000 * number::decode_u32(data)?;
-        let (tp, fsp) = if data.len() >= 2 {
-            (
-                FieldTypeTp::from_u8(data[0]).unwrap_or(FieldTypeTp::Unspecified),
-                data[1],
-            )
-        } else {
-            return Err(Error::unexpected_eof());
-        };
-        *data = &data[2..];
+pub trait TimeDecoder: NumberDecoder {
+    /// `decode_time` decodes time encoded by `encode_time` for Chunk format.
+    fn decode_time(&mut self) -> Result<Time> {
+        let year = i32::from(self.read_u16()?);
+        let buf = self.read_bytes(5)?;
+        let (month, day, hour, minute, second) = (
+            u32::from(buf[0]),
+            u32::from(buf[1]),
+            u32::from(buf[2]),
+            u32::from(buf[3]),
+            u32::from(buf[4]),
+        );
+        let nanoseconds = 1000 * self.read_u32()?;
+        let (tp, fsp) = (
+            FieldTypeTp::from_u8(self.read_u8()?).unwrap_or(FieldTypeTp::Unspecified),
```

Comment:
Could we split apart the tuple? Because the evaluation order of tuple in rust isn't stated explicitly and `read_u8()` has a side-effect, we should not make any assumption on it.

Reply:
After some googling, it's really not stated explicitly.  👍